### PR TITLE
chore: Add a direction setting to the url parameters for internal pages

### PR DESCRIPTION
**Pull request by DaemonCahill**
This PR adds a direction setting to the url parameters for internal pages. A demo of the feature can be seen in the attached video.

https://github.com/cloudscape-design/components/assets/438181/086c82e3-3942-4144-8489-ea77ce500ab9

### CODE REVIEW

File: pages/app/app-context.tsx [hunk 1]
```diff
@@ -8,10 +8,10 @@ import { useHistory, useLocation, useRouteMatch } from 'react-router-dom';
 
 interface AppUrlParams {
   density: Density;
+  direction: string;
```
Comment (pan-kot) [inline pages/app/app-context.tsx:11]: I would change that to a union `"ltr" | "rtl"`

**Review by pan-kot — DISMISSED** [review]

File: src/autosuggest/__integ__/async-autosuggest.test.ts [hunk 2]
```diff
@@ -138,7 +138,7 @@ describe.each([true, false])('Async autosuggest scrolling (with virtualScrolling
     'the bottom of the dropdown lines up with the highlighted option, when going down the list',
     setup({ virtualScrolling }, async page => {
       await page.respondWith(0, true);
-      await page.keys(['ArrowDown', 'ArrowDown', 'ArrowDown']);
+      await page.keys(['ArrowDown', 'ArrowDown', 'ArrowDown', 'ArrowDown', 'ArrowDown', 'ArrowDown']);
```
Comment (just-boris) [inline src/autosuggest/__integ__/async-autosuggest.test.ts:141]: What is happening here and why is unrelated switcher change causes a possible breaking change here?

Comment (pan-kot) [reply src/autosuggest/__integ__/async-autosuggest.test.ts:141]: On smaller screen sizes the header now takes two lines instead of one - that caused some tests that depend on the page height to fail.

Comment (DaemonCahill) [reply src/autosuggest/__integ__/async-autosuggest.test.ts:141]: I was going to suggest removing the header from screenshot tests as well. As far as next steps, who is the appropriate owner of these changes?

File: pages/app/app-context.tsx (hunk 1 above)
Comment (DaemonCahill) [reply pages/app/app-context.tsx]: All set!

**Review by pan-kot — APPROVED** [review]

File: src/autosuggest/__integ__/async-autosuggest.test.ts (hunk 2 above)
Comment (just-boris) [reply src/autosuggest/__integ__/async-autosuggest.test.ts:141]: Let's sync in our team chat and try to see how much we can do within this project

Comment (DaemonCahill) [reply src/autosuggest/__integ__/async-autosuggest.test.ts:141]: For visibility I am going to open a separate PR that handles the demo pages controls in a way that prevents wrapping. Then, we could rebase and merge this PR.

**Comment by DaemonCahill** [comment]
Closing this PR in favor of #1707.
